lighthouse-core/gather/computed/trace-of-tab.js

```diff
@@ -32,6 +32,35 @@ class TraceOfTab extends ComputedArtifact {
     return 'TraceOfTab';
   }
 
+  /**
+   * @param {LH.TraceEvent[]} traceEvents
+   * @param {(e: LH.TraceEvent) => boolean} filter
+   */
+  static filteredStableSort(traceEvents, filter) {
+    // create an array of the indices that we want to keep
+    const indices = [];
+    for (let srcIndex = 0; srcIndex < traceEvents.length; srcIndex++) {
+      if (filter(traceEvents[srcIndex])) {
+        indices.push(srcIndex);
+      }
+    }
+
+    // sort by ts, if there's no ts difference sort by index
+    indices.sort((indexA, indexB) => {
+      const result = traceEvents[indexA].ts - traceEvents[indexB].ts;
+      return result ? result : indexA - indexB;
+    });
+
+    // create a new array using the target indices from previous sort step
+    const sortedArray = new Array(indices.length);
```

Comment:
two inconsistent comments are we fixing micro optimizations or not ;)

Reply:
tested and perf looks pretty much the same either way.

Reply:
> two inconsistent comments are we making micro optimizations or not ;)

ha, I thought about that as I was writing it. In theory it helps, but I only suggested it because it's still idiomatic :)

> tested and perf looks pretty much the same either way.

It might even be a little slower in `filteredStableSort` itself (because it has to realloc as the array grows). In theory the help is elsewhere as array accesses only have to do bounds checks, not hole checks as well.